framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -203,15 +213,38 @@ public Void visitDeclared(AnnotatedDeclaredType type, Void p) {
          * If any constant-value annotation has &gt; MAX_VALUES number of values provided, treats
          * the value as UnknownVal. Works together with ValueVisitor.visitAnnotation, which issues a
          * warning to the user in this case.
+         *
+         * <p>Additional rules applied for IntVal and IntRange annotations: if anno is IntVal and
+         * the number of values provided is &gt; MAX_VALUES, replace with IntRange; if anno is
+         * IntRange and the range is &le; MAX_VALUES, replace with IntVal
          */
-        private void replaceWithUnknownValIfTooManyValues(AnnotatedTypeMirror atm) {
+        private void replaceWithNewAnnoInSpecialCases(AnnotatedTypeMirror atm) {
             AnnotationMirror anno = atm.getAnnotationInHierarchy(UNKNOWNVAL);
 
             if (anno != null && anno.getElementValues().size() > 0) {
-                List<Object> values =
-                        AnnotationUtils.getElementValueArray(anno, "value", Object.class, false);
-                if (values != null && values.size() > MAX_VALUES) {
-                    atm.replaceAnnotation(UNKNOWNVAL);
+                if (AnnotationUtils.areSameByClass(anno, IntRange.class)) {
+                    Range range = getIntRange(anno);
+                    if (range.to < range.from) {
+                        atm.replaceAnnotation(UNKNOWNVAL);
```

Comment:
Can this happen?  I would think the checker should crash in this case, because it indicates that a malformed annotation is being processed.  Quick, visible failure is better than silently suppressing a known problem.